receiver/flinkmetricsreceiver/metadata.yaml

```diff
@@ -0,0 +1,522 @@
+name: flinkmetricsreceiver
+
+resource_attributes:
+  host:
+    description: The host, used to distinguish different jobmanager metrics.
+    type: string
+  taskmanager_id:
+    description: The taskmanager ID, used to distinguish taskmanager metrics.
+    type: string
+  job_name:
+    description: The job name, used to distinguish job metrics.
+    type: string
+  task_name:
+    description: The task name, used to distinguish tasks metrics.
+    type: string
+  subtask_index:
+    description: The subtask index, used to distinguish operator metrics.
+    type: string
+
+attributes:
+  operator_name:
+    description: The operator name, used to distinguish operator metrics.
+    type: string
+  garbage_collector_name:
+    description: The garbage collector name.
+    type: string
+  checkpoint:
+    description: The amount of checkpoints of a given type. 
+    type: string
+    enum: [ in_progress, completed, failed ]
+  record:
+    description: The amount of records of a given type.
+    type: string
+    enum: [ in, out, late_records_dropped ]
+
+metrics:
+  flinkmetrics.jobmanager.status.jvm.cpu.load:
+    enabled: true
+    description: The recent CPU usage of the JVM.
+    unit: 1
+    gauge:
+      value_type: double
+      input_type: string
+    attributes: []
+  flinkmetrics.taskmanager.status.jvm.cpu.load:
+    enabled: true
+    description: The recent CPU usage of the JVM.
+    unit: 1
+    gauge:
+      value_type: double
+      input_type: string
+    attributes: []
+  flinkmetrics.jobmanager.status.jvm.cpu.time:
+    enabled: true
+    description: The CPU time used by the JVM.
+    unit: ns
+    sum:
+      monotonic: true
+      aggregation: cumulative
+      value_type: int
+      input_type: string
+    attributes: []
+  flinkmetrics.taskmanager.status.jvm.cpu.time:
+    enabled: true
+    description: The CPU time used by the JVM.
+    unit: ns
+    sum:
+      monotonic: true
+      aggregation: cumulative
+      value_type: int
+      input_type: string
+    attributes: []
+  flinkmetrics.jobmanager.status.jvm.memory.heap.used:
+    enabled: true
+    description: The amount of heap memory currently used.
+    unit: By
+    sum:
+      monotonic: false
+      aggregation: cumulative
+      value_type: int
+      input_type: string
+    attributes: []
+  flinkmetrics.taskmanager.status.jvm.memory.heap.used:
+    enabled: true
+    description: The amount of heap memory currently used.
+    unit: By
+    sum:
+      monotonic: false
+      aggregation: cumulative
+      value_type: int
+      input_type: string
+    attributes: []
+  flinkmetrics.jobmanager.status.jvm.memory.heap.committed:
+    enabled: true
+    description: The amount of heap memory guaranteed to be available to the JVM.
+    unit: By
+    sum:
+      monotonic: false
+      aggregation: cumulative
+      value_type: int
+      input_type: string
+    attributes: []
+  flinkmetrics.taskmanager.status.jvm.memory.heap.committed:
```

Comment:
Should these be differentiated by attributes instead of having multiple metrics?

Reply:
Possibly. Originally, I did not have the scope type(jobmanager, taskmanager, operator, etc) in the metric name but as an attribute. This made the metric hard follow though because metrics of the same name could have different attributes (the jobmanager require a host but the taskmanagers metrics require both a host and the taskmanager ID since multiple taskmanagers could exist). But now they are listed as resource attributes, so resource metrics are can be applied as needed. The main reason I see not to do it is because it will be easier to distinguish metrics for the user and other solutions metric reporting solutions have the scope type within the metric name as well.

Reply:
Hey @djaglowski , will adding a `flink.scope` resource attribute to clarify if the metric scope belongs to Jobmanager, Taskmanager, Job, Task, or Operator be good? Then I can I can combined the names after one problem is figured out.

So here is the problem by example, if you look at `flink.task.record.count` and `flink.operator.record.count` they have a similar descriptions but have different attributes with the later having `operator_name`. If they are combined, then an empty or a "task" filler would be applied for the `operator_name` for tasks metrics which I don't think is desirable, but I could be wrong and in that case problem solved. 

You may be wondering why the `operator_name` is not a resource attribute and that is because when a single subtask endpoint value is processed, values like metrics like `Sink__Print_to_Std__Out.numRecordsIn` and `Flat_Map.numRecordsIn` from a single response can occur. So when processing, I wait until the end of a loop iteration to emit, but I can emit for every iteration and make it a resource attribute that way. That will however create single `scopeMetrics` values (as seen in integration results in `expected.json`) and a lot them, though I'm not sure if that matters. If doing lots of emits is fine, then I can create `flink.operator.name` as a resource attribute solving the problem.

Reply:
As mentioned elsewhere, I think it's reasonable to have a `flink.resource.type` resource attribute, if it's necessary to apply processors to the data. I think this should be applied only where necessary though. ie. not on every resource just because it's there.

> if you look at flink.task.record.count and flink.operator.record.count they have a similar descriptions but have different attributes

To be clear, different _metric_ attributes would be a good justification to keep these as separate metrics. Different _resource_ attributes would likely not be though.

Reply:
Added `flink.resource.type` for jobmanager and taskmanagers metrics only. Leaving task and operator metrics with the same description as is since only the operator metrics has an operator_name attribute.